alacritty_terminal/src/term/mod.rs

```diff
@@ -1875,8 +1894,16 @@ impl<T: EventListener> Handler for Term<T> {
                     let template = Cell { bg: template.bg, ..Cell::default() };
                     self.grid.clear_viewport(template);
                 }
+
+                self.selection =
+                    self.selection.take().filter(|s| !s.intersects_range(0..=num_lines));
+            },
+            ansi::ClearMode::Saved => {
+                self.grid.clear_history();
+
+                self.selection =
+                    self.selection.take().filter(|s| s.intersects_range(0..=num_lines));
```

Comment:
This isn't correct, right? It should instead be `num_lines + 1 ..= grid.len() - 1`?

Reply:
it's indeed more correct to use what you've wrote + inverse the condition.